package_linter.py

```diff
@@ -437,12 +440,33 @@ def check_deprecated_practices(script):
     if script["name"] == "install" and "ynh_print_info" not in script["shlex"] and "ynh_script_progression" not in script["shlex"]:
         print_warning("Please add a few messages for the user, to explain what is going on (in friendly, not-too-technical terms) during the installation. You can use 'ynh_print_info' or 'ynh_script_progression' for this.")
 
+def install_shellcheck():
```

Comment:
Uh I'm impressed by the effort put into this, but I really don't think that's a good idea :/ 

Some people might want to run this script on a non-debian system (e.g. arch), and not that you implicitly depend from `python-apt` which might not be installed on the machine ... I would rather simply check at the beginning of the script wether or not `shellcheck` is installed, and fail if not ? :s

Reply:
fail to scan with shellcheck or fail any analyze with the script ?

Reply:
Hm I dunno, I guess just failing shellcheck might be less troublesome / frustrating for people trying to launch package_linter for the first time ? I don't remember if you need to install other dependencies or if the script usually work out of the box ..

Reply:
I know the script works with the basic install of yuno, but no more...
Ok for me, I adapt the script to display a warning if the shellcheck test cannot be running.

Reply:
if not present on the system : 
! Binary shellcheck not found in $PATH, try to install it 
! Cannot parse the script with shellcheck

Reply:
the "source" to the file _common.sh for files backup and restore cannot work actually because shellcheck try to find it with a relative path (source ../settings/scripts/_common.sh) and this path not exists before the install of the app...
I search a workaround.

Reply:
OK, I ignore the error "file not found" if the file doesnt' exists. After correction, my package Zabbix in testing pass the check with success.